lib/core/src/persist/receive.rs

```diff
@@ -17,7 +17,7 @@ impl Persister {
 
         let mut stmt = con.prepare(
             "
-            INSERT INTO receive_swaps (
+            INSERT OR REPLACE INTO receive_swaps (
```

Comment:
It seems that these "insert or replace" may override existing onchain data (this is not synced).
For example if get an update for a swap record that already exists in our db. We could have done this if the synced data was separated.

Reply:
Think about the case where there are no local changes (no updated_fields_json) then the process would be:
1. pull the changes
2. apply the record.

Since the applied record doesn't contain on chain data (such as claim id, lockup id) the "insert or replace" will remove them...